eng/Subsets.props

```diff
@@ -249,8 +249,8 @@
       Category="clr" />
   </ItemGroup>
 
-  <!-- Build the CoreCLR cross-arch tools when we're doing a cross-architecture build and either we're building any CoreCLR native tools for platforms CoreCLR fully supports or when someone explicitly requests them -->
-  <ItemGroup Condition="(('$(ClrRuntimeBuildSubsets)' != '' and '$(PrimaryRuntimeFlavor)' == 'CoreCLR' and '$(TargetsMobile)' != 'true') or $(_subset.Contains('+clr.crossarchtools+'))) and '$(BuildArchitecture)' != '$(TargetArchitecture)'">
+  <!-- Build the CoreCLR cross tools when we're doing a cross build and either we're building any CoreCLR native tools for platforms CoreCLR fully supports or when someone explicitly requests them -->
+  <ItemGroup Condition="(('$(ClrRuntimeBuildSubsets)' != '' and '$(PrimaryRuntimeFlavor)' == 'CoreCLR' and '$(TargetsMobile)' != 'true') or $(_subset.Contains('+clr.crossarchtools+'))) and '$(CrossBuild)' == 'true'">
```

Comment:
This is not very obvious but `CrossBuild` property (or `build -cross`) only covers targets which are using `toolchain.cmake` file during cross-compilation. Other targets, such as macOS, Windows and Android, do not set this property during the cross build (e.g. target: osx-arm64, host: osx-x64 bypasses `CrossBuild` and its effect on various subsets build). Therefore, the previous condition is more accurate as we want to build crossarchtools on those targets as well.

Reply:
Currently, I think this is intentional. e.g. `freebsd-x64` and `illumos-x64` builds also use the same prereq images crossdeps layer and run build on `linux-x64`. So we are skipping this build on those platforms.

One of these two options will work:

```diff
- .. and '$(BuildArchitecture)' != '$(TargetArchitecture)'`
+ .. and ('$(BuildArchitecture)' != '$(TargetArchitecture)' or '$(CrossBuild)' != 'true')
```
(preferred; it will not the most scalable solution but it will work if we don't break the `build-rootfs` contract 🙂)

or 
```diff
- .. and '$(BuildArchitecture)' != '$(TargetArchitecture)'`
+ .. and ('$(BuildArchitecture)' != '$(TargetArchitecture)' or '$(HostOS)' != '$(TargetOS)')
```
(tricky because mobile-like targets - and possibly some others - always have a different [`host` and `build`](https://stackoverflow.com/a/15901574/863980) but we don't make this distinction clearly outside `src/mono`)